### DISCUSSION TITLE
Is there anyway to add widget to Screen outside compose function?

### DISCUSSION THREAD

**Discussion by MuongKimhong** [Textualize/textual#3893] [Q&A] [answered]
As far as i know, we can add widget to screen in compose function with `yield` keyword. Is there anyway to add widget outside compose function? For example,

```python
class MyScreen(Screen):
    def compose(self):
        yield SomeWidget()

    def on_some_event(self):
        # add some widget
        yield SomeWidget() # this is not working 
```
For remove, we can do `self.query_one(SomeWidget).remove()`

**Comment by davep** [comment, marked as answer]
Perhaps a quick revisit to the tutorial is in order, [there's a section about that](https://textual.textualize.io/tutorial/#dynamic-widgets).
